crates/block-producer/src/state_view/mod.rs

```diff
@@ -141,7 +152,7 @@ where
 
         // Remove account ids of transactions in block
         for update in block.updated_accounts() {
-            let was_in_flight = locked_accounts_in_flight.remove(&update.account_id());
+            let was_in_flight = locked_accounts_in_flight.remove(&update.account_id()).is_some();
```

Comment:
Is this modification sufficient? Let's say we have two in-flight transactions agains the same account. Let's say the account ID is `a`, the first transaction changes account state from `X` to `Y`, and the second transaction changes account state from `Y` to `Z`. So, the entry in the in-flight accounts map would be `(a, Z)`.

Now, let's say we produce a block which includes only the first transaction. If I'm understanding the code correctly, the entry `(a, Z)` would be removed from the map. So, we'll end up in a situation when the second transaction is still in flight, but there is no entry for the corresponding account in the in-flight accounts map.

Reply:
You're right; I didn't consider partial application of inflight transactions.

I'll handle this using the account state tracker abstraction you recommended.